homeassistant/components/vacuum/xiaomi_miio.py

```diff
@@ -318,6 +340,20 @@ def async_send_command(self, command, params=None, **kwargs):
             "Unable to send command to the vacuum: %s",
             self._vacuum.raw_command, command, params)
 
+    @asyncio.coroutine
+    def async_clean_zone(self,
+                              x1: int = 25500,
+                              y1: int = 25500,
+                              x2: int = 25500,
+                              y2: int = 25500,
+                              repetitions: int = 1):
+        """Clean a specific zone."""
+        params = [[x1, y1, x2, y2, repetitions]]
+        yield from self._try_command(
+            "Unable to start zoned cleaning: %s",
+            self._vacuum.raw_command,
```

Comment:
The proper way to do this would be to fix the python-miio instead of using `raw_command`, see https://github.com/rytilahti/python-miio/pull/311 . It could be also useful to add a service for "goto" for the purpose of emptying the bin :-)

Reply:
https://github.com/rytilahti/python-miio/blob/master/miio/vacuum.py#L80